utils/utils.go

```diff
@@ -17,19 +17,35 @@ limitations under the License.
 package utils
 
 import (
+	"context"
 	"fmt"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"strings"
 	"time"
 
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
+type FatalError struct {
+	Msg string
+}
+
+func (e *FatalError) Error() string {
+	return e.Msg
+}
```

Comment:
- maybe a nit, I particularly don't like this for two reasons.
1. type is created here but returned & used in different pkg
2. at call site, this isn't being asserted efficiently (for want of a better word)
- I propose below, pls give a thought about it (I believe it should work)
---
requirement: signal the caller to os exist when a condition is met
sol: use error value in conjunction with `errors.Is`, something like
``` go
package controllers

import (
  "os"
  "errors"
)

const errRestart = "Restart"

// some sample func
err := func() error { return errRestart }
if errors.Is(err, errRestart) { os.Exit(1) }
```

Reply:
Done